### PR TITLE
MPDX-9801 Populate NS Goal Calculator special needs table and chart

### PR DESCRIPTION
**Pull request by canac**
## Description

- Wire the NS Goal Calculator "Special Needs During MPD" worksheet and chart up to the server-computed cohort-cost figures, replacing the earlier `null`/placeholder UI.
- Add the special-needs fields to the `NewStaffGoalCalculation` GraphQL fragment: `ibsNsoCost`, `faithAndFinanceCost`, `refreshRetreatCost`, `cruConferenceCost`, `specialNeedsSubtotal`, `specialNeedsTotal`, `specialNeedsDevelopedToDate`, `specialNeedsLeft`.
- Add a shared `useSpecialNeedsCategories` hook so the Review-step worksheet table and the Presenting-step chart share the same cohort-cost lines and never drift.
- Refactor `SupportNeedsChart` to be presentational: callers pass a `needsCategories` array (monthly needs or special needs) instead of the chart building rows itself. Adds an `emptyMessage` for the all-zero special-needs case.
- Show the real "Left to Raise" currency value in `NsoInformationSection` (was a `—` placeholder).
- Remove the now-unused `ChartPlaceholderCard`.
- Resolves the `TODO(MPDX-9801)` / `TODO(MPDX-9797)` placeholders now that the calc engine provides these figures.

MPDX-9801 and MPDX-9797

## Testing

- Open the NS Goal Calculator for a calculation with special-needs cohort costs (e.g. `/accountLists/6bbc265f-ec81-4b23-b297-5bd298c4d619/coaching/90935daa-0e49-4b31-a41b-d6b46854ebca/nsGoalCalculator`)
- Go to the **Review Your Calculation** step.
- Check the **Special Needs During MPD** card shows real currency for lines 1–8 (IBS/NSO, Faith & Finances, Refresh Retreat, Cru National Conference, Subtotal, Total, Developed to Date, To be Developed) instead of `—`.
- Go to the **Presenting Your Goal** step.
- Check the **Special Needs Chart** renders a pie of the cohort costs, and shows "No special needs" when all amounts are zero.
- Go to the **NSO Information** section and check **Left to Raise** shows a currency value, not `—`.

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels (_Add the label "Preview" to automatically create a preview environment_)
- [x] I have run the Claude Code `/pr-review` command locally and fixed any relevant suggestions
- [x] I have requested a review from another person on the project
- [x] I have tested my changes in preview or in staging
- [ ] I have cleaned up my commit history

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalStep.test.tsx [hunk 1]
```diff
@@ -136,16 +141,15 @@ describe('PresentingYourGoalStep', () => {
     });
   });
 
-  it('renders the monthly support needs chart and the special needs chart placeholder', async () => {
-    const { findByRole, getByRole, getByTestId } = render(<TestComponent />);
+  it('renders the monthly support needs chart and the special needs chart', async () => {
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalStep.test.tsx:141]: [Medium] This test only asserts `getByRole('heading', { name: 'Special Needs Chart' })`. `PresentationCard` renders that `<h5>` from its `title` prop independent of its children, so this passes even if `SupportNeedsChart` rendered nothing — the old `getByTestId('chart-placeholder')` assertion that verified real chart output was dropped with nothing replacing it. Assert a real legend entry scoped to that card (e.g. `IBS / NSO` or `Administrative Charge`), matching how `SupportNeedsChart.test.tsx` verifies the pie/legend.

File: src/components/HrTools/NsGoalCalculator/Shared/useSpecialNeedsCategories.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  /** Optional worksheet description. */
+  description?: string;
+  amount: number;
+}
+
+/**
+ * Shared by the Special Needs worksheet table (Review step) and the Special
+ * Needs chart (Presenting step) so their labels and amounts never drift.
+ */
+export const useSpecialNeedsCategories = (
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/Shared/useSpecialNeedsCategories.ts:18]: [Medium] New shared hook with no colocated `useSpecialNeedsCategories.test.ts`, unlike its sibling precedent `useMonthlyNeedsRows.test.tsx` in the same feature (Standards Checklist requires a colocated test for every new hook). Indirect coverage via `SpecialNeedsCard.test.tsx` asserts the four amounts/titles, but the category **ordering** and the `description` fields (`Tuition, Books, Housing, Food, Travel Expenses`, `Includes travel`) are asserted nowhere. Add a `renderHook` test asserting order + titles + descriptions + amounts.

File: src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx [hunk 3]
```diff
@@ -74,6 +72,19 @@ export const PresentingYourGoalContent: React.FC<
...
     [married, calculations],
   );
+  const monthlyNeedsRows = useMonthlyNeedsRows(monthlyNeeds);
+  const specialNeeds = useSpecialNeedsCategories(calculations);
+  const specialNeedsCategories = useMemo(
+    () => [
+      ...specialNeeds.map(({ title, amount }) => ({ title, amount })),
+      {
+        title: t('Administrative Charge'),
+        amount:
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx:82]: [Suggestion] The chart's Administrative Charge slice is derived client-side as `specialNeedsTotal - specialNeedsSubtotal`. This reconciles with the worksheet **only if** the server keeps `specialNeedsSubtotal == Σ(lines 1-4)` and `specialNeedsTotal == subtotal / (1 - adminRate)`. Holds today (schema-documented), but the client depends on it silently — if the server ever rounds the subtotal independently, the chart total would drift from the summary card's `specialNeedsTotal` with no error. Consider a short comment documenting the required server relationship, or a server-provided admin-charge field. (Float subtraction itself is fine — display-rounded via `currencyFormat`.)

File: src/components/HrTools/Shared/GoalPresentation/SupportNeedsChart.tsx [hunk 4]
```diff
@@ -55,50 +54,60 @@ const chartColors = [
...
 }) => {
   const locale = useLocale();
   const { containerRef, width } = useContainerWidth();
 
   // In narrower containers, the pie shrinks and the legend moves below the chart
   const isCompact = typeof width === 'number' && width < 700;
 
-  const rows = useMonthlyNeedsRows(monthlyNeeds);
-  const total = rows.reduce((sum, row) => sum + row.amount, 0);
+  const total = needsCategories.reduce((sum, row) => sum + row.amount, 0);
```
Comment (canac) [inline src/components/HrTools/Shared/GoalPresentation/SupportNeedsChart.tsx:82]: [Suggestion] For a staff member with genuinely zero special-needs costs, every amount is `$0.00` so `total` becomes `0`: the pie renders no sectors while the legend still lists all categories (a degenerate empty circle), and the tooltip computes `value / total = 0/0`. `intlFormat`'s finite guard renders that as `0%` (nothing incorrect shows), so this is a UX/robustness nit, not a bug. Consider a "no special needs" empty state when `total === 0`, and add a zero/empty edge-case test for the now-generalized chart.

File: src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx [hunk 5]
```diff
@@ -74,6 +72,19 @@ export const PresentingYourGoalContent: React.FC<
     }),
     [married, calculations],
   );
+  const monthlyNeedsRows = useMonthlyNeedsRows(monthlyNeeds);
+  const specialNeeds = useSpecialNeedsCategories(calculations);
+  const specialNeedsCategories = useMemo(
+    () => [
+      ...specialNeeds.map(({ title, amount }) => ({ title, amount })),
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx]: [Suggestion] `...specialNeeds.map(({ title, amount }) => ({ title, amount }))` strips `description` to conform to `NeedsCategory`, but `SpecialNeedsCategory[]` is already structurally assignable to `NeedsCategory[]` (extra optional `description` is fine). The `.map` is redundant — `...specialNeeds` works directly.

**Review by canac — COMMENTED** [review]
## 🤖 Multi-Agent Code Review — PR #1926

**Verdict: 🔴 BLOCKERS FOUND** · Risk: **CRITICAL (10/10)** · Mode: standard · Agents: Architecture, Testing, Standards, Data Integrity, UX, Financial Reporting + Dependency Impact.

### 🔴 Blocker (verdict driver) — CI-breaking test failure

**`src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalStep.test.tsx:76-78`** · Severity 9.2/10 · **verified by running the suite (`1 failed, 7 passed`)**

This PR intentionally removed `<li>{t('Adjust the Scale to fit on one page.')}</li>` from `PresentingYourGoalContent.tsx`, but the test still asserts it:

```tsx
// PresentingYourGoalStep.test.tsx — delete lines 76-78
-    expect(
-      getByText('Adjust the Scale to fit on one page.'),
-    ).toBeInTheDocument();
```

`grep` confirms the string now lives only in the test. _(Posted in the body rather than as a line comment because line 77 is a context line outside the diff hunks — GitHub's review API can't anchor a comment there. It is still a blocker and drives the BLOCKERS_FOUND verdict.)_

### ✅ What's clean

Zero correctness, security, data-integrity, or financial-calculation defects. The special-needs figures reconcile across all three surfaces (worksheet, chart, summary) under the server contract (verified: `6165+520+750+1200 = 8635`; `8635 / (1-0.12) = 9812.5`). All breaking prop renames (`SupportNeedsChart`, `SpecialNeedsCard`) and the `ChartPlaceholderCard` deletion were fully propagated — no compile break. `NeedsRow.amount` narrowing `number|null → number` is safe (all callers pass non-null `Float!`). Net technical-debt reduction (4 TODO placeholders + 86-line placeholder chart removed).

**CI note:** the `.graphql` fragment change adds 8 `Float!` fields — CI must run `yarn gql` cleanly (generated files aren't committed).

### Medium / Suggestions

See inline comments: missing colocated test for the new `useSpecialNeedsCategories` hook (6.5), a heading-only chart assertion that passes even if the chart renders nothing (6.0), plus low-severity notes on the client-derived Administrative-Charge slice, an all-zero-special-needs empty state, and a redundant `.map`.

### Financial Checklist
Arithmetic safe ✅ · Currency mixing prevented ✅ · Rounding at display boundary only ✅ · Null/undefined handled ✅ · Dates N/A · Server aggregations preferred ✅

---
_Risk is CRITICAL primarily from breadth (15 files) + financial domain, not depth. Fix the blocker; medium/suggestions are optional. Severity <7 items can be dismissed with a `/dismiss: <reason>` reply._

File: src/components/HrTools/NsGoalCalculator/Shared/useSpecialNeedsCategories.ts (hunk 2 above)
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/Shared/useSpecialNeedsCategories.ts:18]: **[Medium] New hook has no colocated test.** _(Testing 6.0 + Architecture 3.5)_

The project Standards checklist requires a colocated `*.test.{ts,tsx}` for every new hook. This hook is deliberately the single source of truth "so their labels and amounts never drift" — exactly the contract worth pinning directly. It's currently covered only indirectly via `SpecialNeedsCard.test.tsx`.

```tsx
// useSpecialNeedsCategories.test.tsx (renderHook, wrap so useTranslation resolves)
it('maps cohort cost fields to labeled rows in order', () => {
  const calculations = { ...defaultGoalCalculation.calculations,
    ibsNsoCost: 6165, faithAndFinanceCost: 520, refreshRetreatCost: 750, cruConferenceCost: 1200 };
  const { result } = renderHook(() => useSpecialNeedsCategories(calculations), { wrapper });
  expect(result.current).toEqual([
    { title: 'IBS / NSO', description: 'Tuition, Books, Housing, Food, Travel Expenses', amount: 6165 },
    { title: 'Faith & Finances Course', amount: 520 },
    { title: 'Refresh Retreat', description: 'Includes travel', amount: 750 },
    { title: 'Cru National Conference', amount: 1200 },
  ]);
});
```

File: src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx [hunk 6]
```diff
@@ -132,19 +142,24 @@ export const PresentingYourGoalContent: React.FC<
...
+            <SupportNeedsChart needsCategories={monthlyNeedsRows} />
           </PresentationCard>
         </Grid>
         <Grid size={{ xs: 12, lg: 6 }}>
-          <ChartPlaceholderCard title={t('Special Needs Chart')} />
+          <PresentationCard
+            title={t('Special Needs Chart')}
+            horizontalScroll={false}
+          >
+            <SupportNeedsChart needsCategories={specialNeedsCategories} />
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx:161]: **[Medium] Special-needs pie amounts are tooltip-only on the Presenting step (a11y).** _(UX)_

The 5 slice amounts (IBS/NSO, Faith & Finance, Refresh, Cru Conference, Administrative Charge) live only in the recharts hover `Tooltip`, which is not keyboard/screen-reader accessible. Unlike the Monthly chart above it — which is backed by `MonthlyNeedsCard`'s per-line breakdown — the sibling `Shared/GoalPresentation/SpecialNeedsCard` on this step renders only a single "Total Special Needs Goal" row. Category *names* are in the legend, so color isn't the sole indicator, and the full per-line table does exist on the Review step.

Maintainer call: mirror the monthly pattern (add a per-category table on this page) or accept the gap as consistent with the pre-existing monthly chart.

File: src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx [hunk 7]
```diff
@@ -74,6 +72,19 @@ export const PresentingYourGoalContent: React.FC<
...
   );
+  const monthlyNeedsRows = useMonthlyNeedsRows(monthlyNeeds);
+  const specialNeeds = useSpecialNeedsCategories(calculations);
+  const specialNeedsCategories = useMemo(
+    () => [
+      ...specialNeeds,
+      {
+        title: t('Administrative Charge'),
+        amount:
+          calculations.specialNeedsTotal - calculations.specialNeedsSubtotal,
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx:83]: **[Suggestion] Admin-charge slice: derived inline, undefended, untested.** _(Financial 3.5 / Architecture 4.0 / Testing 6.0→downgraded)_

`specialNeedsTotal - specialNeedsSubtotal` is safe today — the schema defines line 6 as `subtotal ÷ (1 − adminRate)`, so `total ≥ subtotal` for any non-negative subtotal, and all cost fields are `Float!`. But nothing client-side defends the invariant, and the derived value/order/edge is unasserted. If the server ever returned `subtotal > total`, recharts would draw a negative slice and push other slices' tooltip percentages above 100%.

Optional defensive clamp:
```tsx
amount: Math.max(0, calculations.specialNeedsTotal - calculations.specialNeedsSubtotal),
```
Also note the label `t('Administrative Charge')` is now duplicated with `useMonthlyNeedsRows.ts:69` — consolidating the derivation into `useSpecialNeedsCategories` (returning the derived admin amount alongside the categories) would remove the asymmetry.

**Review by canac — COMMENTED** [review]
## 🤖 Multi-Agent Review — PR #1926

**Verdict: ✅ APPROVED WITH SUGGESTIONS** · **0 blockers** · Risk 10/10 (CRITICAL — breadth-driven, not danger)

6 specialized agents (Financial Reporting, Data Integrity, Architecture, Testing, UX, Standards) + dependency-impact analysis, standard mode. All checks green: `yarn gql` / `yarn lint:ts` / `eslint` exit 0, 14/14 touched tests pass. Clean placeholder→real-value wiring with net technical-debt reduction (deleted `ChartPlaceholderCard`, 4 TODOs, and a null-branch). The `SupportNeedsChart` prop break (`monthlyNeeds`→`needsCategories`) is fully contained to its two updated callers; no dangling refs.

Nothing blocks merge. Inline comments cover **2 medium** findings + suggestions.

### ⚠️ Deploy coordination
The 8 new `NewStaffGoalCalculation` fields are **non-nullable** in the codegen schema. If prod `api.mpdx.org` does not yet return them, the entire `newStaffGoalCalculation` query **hard-errors** (non-null violation) rather than silently showing `$0.00`. Confirm the API side ships with/before this client change.

### Additional low-severity notes (not line-anchored)
- **Print page-break (UX, 3.5):** the deleted `ChartPlaceholderCard` set `@media print { pageBreakInside: 'avoid' }`; the shared `SupportNeedsChart` print styles (`SupportNeedsChart.tsx:16-34`) don't, so the special-needs pie can now split across a page break (mitigated by the surrounding `zoom: 0.4`). Consider adding it so both charts benefit.
- **Type proliferation (Architecture, 3.0):** `NeedsCategory` / `SpecialNeedsCategory` / `NeedsRow` / `SupportNeedsRow` structurally overlap; cosmetic, optional `extends`.

### Debate note
Testing initially rated the `SupportNeedsChart` zero-total path **6.5** ("renders NaN%/∞%"). Refuted by source and downgraded: `percentageFormat` (`src/lib/intlFormat.ts:32`) coerces non-finite → `0`, and `currencyFormat` (`:47`,`:58`) likewise, so an all-zero/empty dataset renders "0%", never "NaN%". The surviving point is a missing edge-case *test*, not a user-facing defect.

<sub>Reviewer level: SENIOR. CRITICAL risk ⇒ not auto-approve eligible; human sign-off expected regardless of verdict.</sub>

File: src/components/HrTools/Shared/GoalPresentation/SupportNeedsChart.tsx (hunk 4 above)
Comment (canac) [inline src/components/HrTools/Shared/GoalPresentation/SupportNeedsChart.tsx:82]: **[Suggestion] Empty/zero + single-item boundary cases untested.** _(Testing)_

`total = needsCategories.reduce(...)` then `Number(value) / total` in the tooltip. Div-by-zero is **neutralized** at the format boundary — `percentageFormat` (`src/lib/intlFormat.ts:32`) coerces non-finite → 0, so it renders "0%", not "NaN%" (verified; this is why the finding is a suggestion, not a defect). But the repo convention of covering 0/1/many for a shared component is unmet — `SupportNeedsChart.test.tsx` only exercises the many-item happy path. Add an empty-array case and an all-zero case.

File: src/components/HrTools/NsGoalCalculator/GoalSettings/Sections/NsoInformationSection.tsx [hunk 8]
```diff
@@ -81,9 +86,13 @@ export const NsoInformationSection: React.FC<GoalSettingsSectionProps> = ({
...
       </FieldRow>
 
-      {/* TODO(MPDX-9797): Computed value — wire up once the calc engine lands. */}
       <FieldRow label={t('Left to Raise')}>
-        <GoalSettingsPlaceholder label={t('Left to Raise')} value="—" />
+        <Typography variant="body1">
+          <Box component="span" sx={visuallyHidden as SxProps<Theme>}>
+            {t('Left to Raise')}
+          </Box>
+          {formatCurrency(calculations.specialNeedsLeft)}
```
Comment (canac) [inline src/components/HrTools/NsGoalCalculator/GoalSettings/Sections/NsoInformationSection.tsx:90]: **[Suggestion] "Left to Raise" now shows real currency instead of the `—` placeholder.** _(Financial)_

This is the intended change and correct — `specialNeedsLeft` is `Float!` (documented never-negative), so `$0.00` legitimately means "fully funded," not "unknown." The only residual is a product/contract dependency: confirm the server never emits an "unpopulated" `0` here (e.g. before the calc is computed) that would masquerade as a real figure. No code change needed.

File: src/components/HrTools/NsGoalCalculator/Shared/useSpecialNeedsCategories.ts (hunk 2 above)
Comment (canac) [reply src/components/HrTools/NsGoalCalculator/Shared/useSpecialNeedsCategories.ts:18]: /dismiss Trivial hook tested through the components that use it

File: src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx (hunk 7 above)
Comment (canac) [reply src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx:83]: /dismiss I accept relying on the server calculations being stable.

File: src/components/HrTools/NsGoalCalculator/GoalSettings/Sections/NsoInformationSection.tsx [hunk 9]
```diff
@@ -81,9 +86,13 @@ export const NsoInformationSection: React.FC<GoalSettingsSectionProps> = ({
         />
       </FieldRow>
 
-      {/* TODO(MPDX-9797): Computed value — wire up once the calc engine lands. */}
       <FieldRow label={t('Left to Raise')}>
-        <GoalSettingsPlaceholder label={t('Left to Raise')} value="—" />
+        <Typography variant="body1">
+          <Box component="span" sx={visuallyHidden as SxProps<Theme>}>
+            {t('Left to Raise')}
+          </Box>
```
Comment (zweatshirt) [inline src/components/HrTools/NsGoalCalculator/GoalSettings/Sections/NsoInformationSection.tsx]: This may be redundant, since a screenreader should be able to read the label on the `FieldRow`. I could be mistaken but this would cause a double announcement. Worth confirming.

File: src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx (hunk 6 above)
Comment (zweatshirt) [inline src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx:161]: Optional: Do want to consider the empty state for the Special Needs chart? I noticed that when the fields are all 0s it renders an empty legend. I'm not sure how feasible that path is in production, however. I think it might happen with scenario goals.

**Review by zweatshirt — APPROVED** [review]
Looks awesome, nice to see it all come together. Thanks Caleb. Just had a few minor concerns.

File: src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx (hunk 6 above)
Comment (canac) [reply src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx:161]: I opted not to have an empty state because scenario goals don't have a Presenting Your Goal page. Do you think I should add one?

File: src/components/HrTools/NsGoalCalculator/GoalSettings/Sections/NsoInformationSection.tsx (hunk 9 above)
Comment (canac) [reply src/components/HrTools/NsGoalCalculator/GoalSettings/Sections/NsoInformationSection.tsx]: I think you're right if someone navigates the page element by element

File: src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx (hunk 6 above)
Comment (zweatshirt) [reply src/components/HrTools/NsGoalCalculator/PresentingYourGoal/PresentingYourGoalContent.tsx:161]: Ah good point, no worries then
